Kinvey.Android/Push/KinveyFCMService.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Android.App;
+using Android.Content;
+using Android.OS;
+
+namespace Kinvey
+{
+    [Service]
+    public abstract class KinveyFCMService : IntentService
+    {
+        public KinveyFCMService()
+        { }
+
+        static PowerManager.WakeLock sWakeLock;
+        static object LOCK = new object();
+        private const string MESSAGE_FROM_FCM = "gcm.notification.body";
+        private const string REGISTRATION_ID = "registration_id";
+        private const string KINVEY_FCM = "KinveyFCM";
+
+        protected override void OnHandleIntent(Intent intent)
+        {
+
+            lock (LOCK)
+            {
+                if (sWakeLock == null)
+                {
+                    var pm = PowerManager.FromContext(this.ApplicationContext);
+                    sWakeLock = pm.NewWakeLock(WakeLockFlags.Partial, KINVEY_FCM);
```

Comment:
@romanogolikhin why do we need to keep the device on while you are handling the intent?

Reply:
right... worth it to do some research about it. @vinaygahlawat will check how this is done in the android sdk. hold on for now.